cmd/libsnap-confine-private/mount-opt.c

```diff
@@ -115,3 +117,122 @@ const char *sc_mount_opt2str(unsigned long flags)
 	}
 	return buf;
 }
+
+static void sc_grow_string(char **s, const char *extra)
```

Comment:
I'm slightly concerned about the direction we are talking with snap-confine. It is a suid root binary and we want it to be small and easy. Functions like this make me uneasy. It feels like we re-inventing low-level library. Sorry for being so negative but I think we should think about alternatives, maybe just using simpler logging?